ndc-models/src/lib.rs

```diff
@@ -704,6 +704,16 @@ pub enum Field {
         #[serde(skip_serializing_if = "BTreeMap::is_empty", default)]
         arguments: BTreeMap<ArgumentName, Argument>,
     },
+    NestedCollection {
+        /// The name of the column which contains the nested data
+        column: FieldName,
+        /// Column arguments
+        arguments: BTreeMap<ArgumentName, Argument>,
+        /// Path to a nested array of objects contained within the selected column
+        field_path: Option<Vec<FieldName>>,
```

Comment:
Do we need `field_path`, or could we use `Field::NestedField` to allow drilling through object paths? So instead of 

```yaml
CustomerAddresses:
  type: NestedCollection
  column: Customer
  arguments: {}
  field_path: ["Addresses"]
  query:
    aggregates:
      countryCount:
        type: ColumnCount
        column: Country
        distinct: true
```

we did
```yaml
Customer:
  type: Column
  column: Customer
  fields:
    type: Object
    fields:
      Addresses:
        type: NestedCollection
        column: Addresses
        arguments: {}
        query:
          aggregates:
            countryCount:
              type: ColumnCount
              column: Country
              distinct: true
```

Feels like the response shape more closely matches the nested object structure when it's done like this.

The other places we use `field_path` we don't actually care about the nesting because we're just using the result of traversing the nesting in a computation (eg predicates, ordering). But here, we do.

Reply:
Good point, thanks, I don't think we need `field_path`.